CUDADataFormats/Common/interface/CUDAProductBase.h

```diff
@@ -36,11 +36,19 @@ class CUDAProductBase {
 
   int device() const { return device_; }
 
-  const cuda::stream_t<>& stream() const { return *stream_; }
-  cuda::stream_t<>& stream() { return *stream_; }
-
-  const cuda::event_t* event() const { return event_.get(); }
-  cuda::event_t* event() { return event_.get(); }
+  // cudaStream_t is a pointer to a thread-safe object, for which a
+  // mutable access is needed even if the CUDAScopedContext itself
+  // would be const. Therefore it is ok to return a non-const
+  // pointer from a const method here.
+  cudaStream_t stream() const { return stream_->id(); }
+  cudaStream_t stream() { return stream_->id(); }
```

Comment:
why do we need the non-`const` method, here and below ?

Reply:
No good reason, seems like a too blind transformation.